modules/core/src/controllers/transition-manager.js

```diff
@@ -45,7 +47,7 @@ export default class TransitionManager {
     let transitionTriggered = false;
     const currentProps = this.props;
     // Set this.props here as '_triggerTransition' calls '_updateViewport' that uses this.props.
-    nextProps = Object.assign({}, this.props, nextProps);
+    nextProps = Object.assign({}, DEFAULT_PROPS, nextProps);
```

Comment:
Wouldn't this trigger invalid transition in following cases :
```
DEFAULT_PROPS  = {a: 1, b: 1}
this.props = {a: 10, b: 10}
passed in nextProps = {a: 20}
```
triggers transition between {a:10 b: 10} to {a:20, b: 1} , but actually we should trigger transition {a: 10, b:10} to {a: 20, b: 10}